lib/reservation.rb

```diff
@@ -0,0 +1,39 @@
+require "date"
+
+module Hotel
+  class Reservation
+    attr_accessor :start_date, :end_date, :total_cost, :room
+
+    ROOM_COST = 200 # identical rooms at $200/night
+
+    def initialize(start_date, end_date, room)
+      @start_date = start_date
+      @end_date = end_date
+      @room = room
+
+      if !room.to_s.match /\d+/
+        raise ArgumentError, "Invalid Room Number"
+      end
+
+      if start_date >= end_date
+        raise ArgumentError, "end date must be after start date. \
+        Start date: #{start_date}, end date: #{end_date}"
+      end
+    end
+
+    def total_cost
+      return ROOM_COST * total_nights
+    end
```

Comment:
I like the decision to make this a method rather than storing the cost as an instance variable. That way if the number of nights or the cost per night were to change (say in a future version of the program), the total cost will automatically be correct.